src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotDao.java

```diff
@@ -42,19 +44,21 @@ public TSnePlotDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
     }
 
     private static final String SELECT_T_SNE_PLOT_WITH_CLUSTERS_STATEMENT =
-            "SELECT tsne.cell_id, tsne.x, tsne.y, clusters.cluster_id " +
-            "FROM scxa_tsne AS tsne " +
-                "LEFT JOIN " +
-                "(SELECT * FROM scxa_cell_clusters WHERE k=:k) AS clusters " +
-                "ON clusters.cell_id=tsne.cell_id AND clusters.experiment_accession=tsne.experiment_accession " +
-            "WHERE tsne.experiment_accession=:experiment_accession AND tsne.perplexity=:perplexity";
+			"SELECT tsne.cell_id, tsne.x, tsne.y, clusters.cluster_id" +
+					"            FROM scxa_tsne AS tsne" +
+					"                LEFT JOIN" +
+					"                (SELECT mem.*, g.value as cluster_id FROM scxa_cell_group_membership as mem join  scxa_cell_group g" +
+					"                    on mem.cell_group_id = g.id and g.variable = :k) AS clusters" +
+					"                ON clusters.cell_id=tsne.cell_id AND clusters.experiment_accession=tsne.experiment_accession" +
```

Comment:
SInce we are not using it inside the subquery
```suggestion
					"                ON clusters.cell_id=tsne.cell_id " +
```